### PR TITLE
[Gym] Ignore unnessesary Provisioning Profiles

### PR DESCRIPTION
**Pull request by thasegaw**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context
Gym collect all defined Provisioning Profiles in Xcode.
But we need only it set in Gym Configuration options.

### Description
This PR allow Gym not to collect unused Provisioning Profiles.

#### before
```
$ bundle exec fastlane -v
fastlane installation at path:
/Users/thasegaw/work/app/topapp/vendor/bundle/ruby/2.3.0/gems/fastlane-2.49.0/bin/fastlane
-----------------------------
fastlane 2.49.0

$ bundle exec fastlane gym --workspace Sample.xcworkspace --scheme Sample --configuration InHouse --export_method enterprise
[10:00:36]: Detected provisioning profile mapping: {"jp.co.sample.appli.debug"=>"Sample Development", "jp.co.sample.appli.test"=>"SampleTest Development", "jp.co.sample.appli"=>"Sample Development", "jp.co.sample.appli.staging"=>"Sample Development", "jp.co.sample.appli.qa"=>"Sample Development", "jp.co.sample.appli.debug.SampleToday"=>"Sample Development", "jp.co.sample.appli.test.SampleToday"=>"Sample Development", "jp.co.sample.appli.SampleToday"=>"Sample Development", "jp.co.sample.appli.staging.SampleToday"=>"Sample Development", "jp.co.sample.appli.qa.SampleToday"=>"Sample Development", "jp.co.sample.appli.debug.SampleNotificationService"=>"Sample Development", "jp.co.sample.appli.test.SampleNotificationService"=>"Sample Development", "jp.co.sample.appli.SampleNotificationService"=>"Sample Development", "jp.co.sample.appli.staging.SampleNotificationService"=>"Sample Development", "jp.co.sample.appli.qa.SampleNotificationService"=>"Sample Development", "jp.co.sample.appli.debug.SampleNotificationContent"=>"Sample Development", "jp.co.sample.appli.test.SampleNotificationContent"=>"Sample Development", "jp.co.sample.appli.SampleNotificationContent"=>"Sample Development", "jp.co.sample.appli.staging.SampleNotificationContent"=>"Sample Development", "jp.co.sample.appli.qa.SampleNotificationContent"=>"Sample Development"}
```
#### after
```
fastlane gym --workspace Sample.xcworkspace --scheme Sample --configuration InHouse --export_method enterprise
[09:57:34]: Detected provisioning profile mapping: {"jp.co.sample.appli"=>"Sample Development", "jp.co.sample.appli.SampleToday"=>"Sample Development", "jp.co.sample.appli.SampleNotificationService"=>"Sample Development", "jp.co.sample.appli.SampleNotificationContent"=>"Sample Development"}
```

### CODE REVIEW

**Comment by KrauseFx** [comment]
Thanks for the PR 👍  Is _gym_ not working if it detects too many provisioning profiles, or is it just not looking nicely?

**Comment by thasegaw** [comment]
@KrauseFx Gym works well without this PR. 
This PR is intended to make Gym better behavior.

File: gym/lib/gym/detect_values.rb [hunk 1]
```diff
@@ -94,6 +98,8 @@ def self.detect_selected_provisioning_profiles
           project = Xcodeproj::Project.open(project_path)
           project.targets.each do |target|
             target.build_configuration_list.build_configurations.each do |build_configuration|
+              next unless build_configuration.to_s == Gym.config[:configuration]
```
Comment (ohayon) [inline gym/lib/gym/detect_values.rb]: What does this line do?

**Review by ohayon — COMMENTED** [review]
Hey @thasegaw - Thanks for this PR it looks like a great change! I just have one question about a line above. Would love to hear your thoughts! 😄

File: gym/lib/gym/detect_values.rb (hunk 1 above)
Comment (thasegaw) [inline gym/lib/gym/detect_values.rb]: It indicates that `Gym` skip unused configuration.
If I set `configuration: 'Debug'` in Gym option, other configration (like `Release`) is unnessesary to archive package.

Comment (KrauseFx) [inline gym/lib/gym/detect_values.rb]: I'm not 100% sure about that part, the rest looks great 👍

**Comment by KrauseFx** [comment]
Hey @thasegaw, sorry for the late reply, could you resolve the merge conflicts, and remove the configuration check for now, let's stick with removing the test targets only 👍

**Comment by thasegaw** [comment]
I fixed it!

**Review by KrauseFx — APPROVED** [review]
This is awesome, thank you for submitting and updating the PR 🚀

**Pull request merged**